jpype/_jarray.py

```diff
@@ -92,6 +92,18 @@ def __new__(cls, tp, dims=1):
     def of(cls, array, dtype=None):
         return _jpype.arrayFromBuffer(array, dtype)
 
+    def __class_getitem__(cls, key):
+        if key is _jpype.JClass:
+            # explicit check for JClass
+            # _toJavaClass cannot be used
+            # passing int, float, etc is not allowed
+            key = _jpype._java_lang_Class
+        if isinstance(key, (str, _jpype._java_lang_Class)):
+            key = _jpype.JClass(key)
+        if isinstance(key, _jpype.JClass):
+            return type(key[0])
+        raise TypeError("Cannot instantiate unspecified array type")
+
```

Comment:
Was this path tested?   I would worry that the code above didn't check for the case and we may segfault.

Reply:
I would argue for the exception message to be changed. As far as I can tell, it is impossible to attempt to instantiate an unspecified array type.

I left the exception message as it was to discuss later.


I'm not sure how to tell the code coverage that most of the paths are covered. I did miss two paths which I'll add tonight.

Reply:
Regarding the code coverage, it was actually correct and I was missing coverage due to a copy/paste mistake.